io/subsystem/src/main/java/org/wildfly/extension/io/WorkerAdd.java

```diff
@@ -189,7 +221,14 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             }
         }
 
-        final WorkerService workerService = new WorkerService(builder.getMap());
+        OptionMap options = builder.getMap();
+        final ThreadsService threadsService = new ThreadsService(options.get(Options.WORKER_TASK_MAX_THREADS));
```

Comment:
What is idea behind this? just have a capability that provides read-only view of max threads that worker has?